src/applications/disability-benefits/all-claims/submit-transformer.js

```diff
@@ -264,6 +264,13 @@ export function transform(formConfig, form) {
     return formData;
   };
 
+  const removeRedundantRetirementPayQuestion = formData => {
+    /* eslint-disable no-param-reassign */
+    delete formData.hasMilitaryRetiredPay;
+    /* eslint-enable no-param-reassign */
```

Comment:
[See this comment about mutation in JS](https://github.com/department-of-veterans-affairs/vets-website/pull/39955/files#r2529585878)